jscomp/ml/typecore.ml

```diff
@@ -2983,8 +3027,12 @@ and type_label_access env srecord lid =
   let ty_exp = record.exp_type in
   let opath =
     try
-      let (p0, p, _, _) = extract_concrete_record env ty_exp in
-      Some(p0, p)
+      match extract_concrete_typedecl env ty_exp with
+      | (p0, _, {type_attributes}) 
+        when Path.same p0 Predef.path_dict && Dicts.has_dict_attribute type_attributes -> 
```

Comment:
not sure I get what case this covers, but I'll see in the tests I guess

Reply:
I'll add a comment.